coriolis/tests/minion_manager/rpc/data/get_minion_pool_refresh_flow.yaml

```diff
@@ -0,0 +1,159 @@
+      db_calls:
+        include:
+          - set_minion_machine_allocation_status: 
+              id: minion1
+              allocation_status: POWERING_OFF
+          - set_minion_machine_allocation_status:
+              id: minion3
+              allocation_status: HEALTHCHECKING
+          - set_minion_machine_allocation_status:
+              id: minion4
+              allocation_status: HEALTHCHECKING
+        exclude:
+          - set_minion_machine_allocation_status:
+              id: minion2
+              allocation_status: HEALTHCHECKING
+          - set_minion_machine_allocation_status:
+              id: minion2
+              allocation_status: POWERING_OFF
+
+# delete machines that are not needed
+- config:
+    minion_pool:
+      id: minion_pool1
+      # minimum_minions 1 means we have too many minions
+      minimum_minions: 1
+      minion_max_idle_time: 0
+      minion_retention_strategy: delete
+      minion_machines:
+        - id: minion1
+          allocation_status: AVAILABLE
+          power_status: POWERING_OFF
+        - id: minion2
+          allocation_status: ERROR
+          power_status: POWERED_ERROR
+        - id: minion3
+          allocation_status: AVAILABLE
+          power_status: POWERED_ON    
+        - id: minion4
+          allocation_status: ERROR_DEPLOYING
+          power_status: POWERED_ON    
+  expect:
+    result:
+      flow_tasks: 
+        - pool-minion_pool1-machine-minion1-deallocation
+        - pool-minion_pool1-machine-minion3-healthcheck
+        - pool-minion_pool1-machine-minion4-healthcheck
+      db_calls:
+        include:
+          - set_minion_machine_allocation_status: 
+              id: minion1
+              allocation_status: DEALLOCATING
+          - set_minion_machine_allocation_status:
+              id: minion3
+              allocation_status: HEALTHCHECKING
+          - set_minion_machine_allocation_status:
+              id: minion4
+              allocation_status: HEALTHCHECKING
+        exclude:
+          - set_minion_machine_allocation_status:
+              id: minion2
+              allocation_status: HEALTHCHECKING
+          - set_minion_machine_allocation_status:
+              id: minion2
+              allocation_status: DEALLOCATING
+
+# invalid retention strategy
+- config:
+    minion_pool:
+      id: minion_pool1
+      # minimum_minions 1 means we have too many minions
+      minimum_minions: 1
+      minion_max_idle_time: 0
+      minion_retention_strategy: invalid
+      minion_machines:
+        - id: minion1
+          allocation_status: AVAILABLE
+          power_status: POWERING_OFF
+        - id: minion2
+          allocation_status: ERROR
+          power_status: POWERED_ERROR
+        - id: minion3
+          allocation_status: AVAILABLE
+          power_status: POWERED_ON
+        - id: minion4
+          allocation_status: ERROR_DEPLOYING
+          power_status: POWERED_ON
+  expect:
+    exception: InvalidMinionPoolState
```

Comment:
Please add a case where no machines get healthchecked, in order to reach this: https://github.com/cloudbase/coriolis/blob/master/coriolis/minion_manager/rpc/server.py#L1425-L1429